### PR TITLE
Export sites only vcf STEP 1-- 317 add AC, AN, AF to the final VCF

### PR DESCRIPTION
**Pull request by RoriCremer**
So... AC, AN and AF annotations are back in the final VCF.

A few things to note:
These do not respect the FT tag---working with Louis and Laura to make that happen since Louis feels the change will have to happen in htsjdk itself (specifically in the getCalledChrCount method in VariantContext and its different versions)

This solution doesn't pass in any founderIds--likely I'm just missing where they should get pulled from? I assume there are people in AoU that are related? Or is that way outside the scope of any of this?

We may want to add a boolean to make the sites-only VCF require less cleanup from the user.

example is from manually running:

 ./gatk --java-options "-Xmx9g"             ExtractCohort                 --mode GENOMES --ref-version 38                 -R "/Users/aurora/projects/references/hg38/v0/Homo_sapiens_assembly38.fasta"                 -O "addannotations.vcf.gz"   --local-sort-max-records-in-ram 10000000  --sample-table "spec-ops-aou.anvil_100_for_testing.sample_info"  --project-id "spec-ops-aou"         --cohort-extract-table "spec-ops-aou.anvil_100_for_testing.Rori_Test"   -L chr20:1-1000000

Screen Shot 2021-05-30 at 12 55 06 PM

### CODE REVIEW

**Review by ahaessly — APPROVED** [review]
👍

File: src/main/java/org/broadinstitute/hellbender/tools/gvs/extract/ExtractCohortEngine.java [hunk 1]
```diff
@@ -33,6 +33,8 @@
 import java.util.*;
 import java.util.stream.Collectors;
 
+import static htsjdk.variant.variantcontext.VariantContextUtils.calculateChromosomeCounts;
```
Comment (kcibul) [inline src/main/java/org/broadinstitute/hellbender/tools/gvs/extract/ExtractCohortEngine.java:36]: I find static imports a bit hard to follow in the code, not sure what the GATK style is though (but I haven't seen it very often)

File: src/main/java/org/broadinstitute/hellbender/tools/gvs/extract/ExtractCohortEngine.java [hunk 2]
```diff
@@ -518,6 +521,18 @@ protected VariantContext removeAnnotations(VariantContext vc) {
...
+    }
+
+    protected VariantContext addChromosomeCountsAnnotations(VariantContext vc) {
+
+        final VariantContextBuilder builder = new VariantContextBuilder(vc);
+
+        // add AC/AF/AD annotations
+        Set<String> founderIds = new HashSet<>();
+        Map<String, Object> attributes = new HashMap<>(vc.getAttributes());
+        Map<String, Object> chromosomeCounts = calculateChromosomeCounts(vc, attributes, true, founderIds);
```
Comment (kcibul) [inline src/main/java/org/broadinstitute/hellbender/tools/gvs/extract/ExtractCohortEngine.java:534]: I think there's a form of this call that doesn't take the last parameter, might bel more clear

File: src/main/java/org/broadinstitute/hellbender/tools/gvs/extract/ExtractCohortEngine.java [hunk 3]
```diff
@@ -518,6 +521,18 @@ protected VariantContext removeAnnotations(VariantContext vc) {
...
+        builder.putAttributes(chromosomeCounts);
```
Comment (kcibul) [inline src/main/java/org/broadinstitute/hellbender/tools/gvs/extract/ExtractCohortEngine.java:535]: the docs on this are confusing (about removing stale attributes, why we pass things in and out, etc) so I'm not sure we need to call putAttribues here...

but the bigger question would be, why not just use the GATK annotation infrastructure?

**Review by kcibul — COMMENTED** [review]
I think in the tests we are still running Gnarly (which outputs these attributes) so I'm not sure which output we are seeing... Can you disable gnarly for the integration test and make sure things are still passing?
